File: Python/5kyu/pig_it.py
# Simple Pig Latin
# https://www.codewars.com/kata/520b9d2ad5c005041100000f

# My code
import re
def pig_it(text):
    s = ''
    for i in text.split():
        s += i[1:] + i[0] + 'ay ' if re.match('[a-z,A-Z]', i) else i + ' '
    return s[:-1]

# Best code
# def pig_it(text):
#     lst = text.split()
#     return ' '.join( [word[1:] + word[:1] + 'ay' if word.isalpha() else word for word in lst])

File: JavaScript/5kyu/pigIt.js
// Simple Pig Latin
// https://www.codewars.com/kata/520b9d2ad5c005041100000f

// My code
function pigIt(str){
  let s = '';
  for (let i of str.split(' ')){
    s += (/[a-z]/i.test(i)) ? `${i.slice(1, i.length)}${i.slice(0, 1)}ay ` : `${i} `
  }
  return s.slice(0, s.length - 1)
}

// Best code
// function pigIt(str){
//   return str.replace(/(\w)(\w*)(\s|$)/g, "\$2\$1ay\$3")
// }